fix the install error: make: execvp: /bin/sh: Argument list too long

diff --git a/Makefile b/Makefile
index 983d120..f74fa98 100644
--- a/Makefile
+++ b/Makefile
@@ -2982,7 +2982,7 @@ install-headers_c:
 install-headers_cxx:
 	$(E) "[INSTALL] Installing public C++ headers"
 	$(Q) $(foreach h, $(PUBLIC_HEADERS_CXX), $(INSTALL) -d $(prefix)/$(dir $(h)) && ) exit 0 || exit 1
-	$(Q) $(foreach h, $(PUBLIC_HEADERS_CXX), $(INSTALL) $(h) $(prefix)/$(h) && ) exit 0 || exit 1
+	$(Q) for h in $(PUBLIC_HEADERS_CXX); do $(INSTALL) "$$h" $(prefix)/$$h || exit 1; done
 
 install-static: install-static_c install-static_cxx
 
